### PR TITLE
[5.2RC3] fix to exclude current article even in include mode

### PR DESCRIPTION
**Pull request by rdeutz**
### Summary of Changes
With this change we can also use the exclude current article when we chose include article ids

### Testing Instructions
Try to exclude some articles and include some articles. Check if the module show the expected articles. You should be able to use the exclude current article and include article at the same time. To test you have to add articles to the list of included articles and navigate to one of the included articles.

### Expected result AFTER applying this Pull Request
Module shows the expected articles

### CODE REVIEW

File: modules/mod_articles/mod_articles.xml [hunk 1]
```diff
@@ -105,30 +105,28 @@
...
+				<field
+					name="ex_or_include_articles"
+					type="list"
+					label="MOD_ARTICLES_FIELD_EX_OR_INCLUDE_LABEL"
+					default="0"
+					filter="integer"
+				>
+					<option value="0">MOD_ARTICLES_OPTION_EXCLUDE_VALUE</option>
+					<option value="1">MOD_ARTICLES_OPTION_INCLUDE_VALUE</option>
+				</field>
```
Comment (Quy) [inline modules/mod_articles/mod_articles.xml]:
```suggestion
					>
					<option value="0">JNO</option>
					<option value="1">JYES</option>
				</field>

				<field
					name="ex_or_include_articles"
					type="list"
					label="MOD_ARTICLES_FIELD_EX_OR_INCLUDE_LABEL"
					default="0"
					filter="integer"
					>
					<option value="0">MOD_ARTICLES_OPTION_EXCLUDE_VALUE</option>
					<option value="1">MOD_ARTICLES_OPTION_INCLUDE_VALUE</option>
				</field>
```

Comment (rdeutz) [reply modules/mod_articles/mod_articles.xml]: always the same I forget about the xml, stupid me.

**Comment by obuisard** [comment]
I have tested this item :white_check_mark: successfully on 007ff98ad69475e66941f291bf4e7baad4a98907<br /><br />Works as expected.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/44213">issues.joomla.org/tracker/joomla-cms/44213</a>.</sub>

File: modules/mod_articles/src/Helper/ArticlesHelper.php [hunk 2]
```diff
@@ -227,16 +228,40 @@ public function getArticles(Registry $params, SiteApplication $app)
...
 
             $articlesListToProcess = $params->get('excluded_articles', '');
         }
 
         foreach (ArrayHelper::fromObject($articlesListToProcess) as $article) {
+            if (
+                $ex_or_include_articles === 1
+                && $params->get('exclude_current', 1) === 1
+                && $input->get('option') === 'com_content'
+                && $input->get('view') === 'article'
```
Comment (Quy) [inline modules/mod_articles/src/Helper/ArticlesHelper.php]:
```suggestion
                $ex_or_include_articles === 1
                && $params->get('exclude_current', 1) === 1
                && $input->get('option') === 'com_content'
                && $input->get('view') === 'article'
```
These checks can be assigned to a variable above the loop to be executed once.  Then no need to do the same checks except with the id in the loop. This variable can be used again in the next if statement.

Comment (rdeutz) [reply modules/mod_articles/src/Helper/ArticlesHelper.php]: For now I will let it as it is, we need to get it in

File: modules/mod_articles/src/Helper/ArticlesHelper.php [hunk 3]
```diff
@@ -227,16 +228,40 @@ public function getArticles(Registry $params, SiteApplication $app)
...
         }
 
+        // Edge case when the user select include mode but didn't add an article,
+        // we might have to exclude the current article
+        if (
+            $ex_or_include_articles === 1
+            && $params->get('exclude_current', 1) === 1
+            && $input->get('option') === 'com_content'
+            && $input->get('view') === 'article'
+            && (int) $article['id'] === $currentArticleId
```
Comment (Quy) [inline modules/mod_articles/src/Helper/ArticlesHelper.php]: > Warning: Undefined variable $article in \modules\mod_articles\src\Helper\ArticlesHelper.php on line 258

Comment (rdeutz) [reply modules/mod_articles/src/Helper/ArticlesHelper.php]: shit but fixed

File: modules/mod_articles/src/Helper/ArticlesHelper.php [hunk 4]
```diff
@@ -216,27 +216,48 @@ public function getArticles(Registry $params, SiteApplication $app)
         $ex_or_include_articles = $params->get('ex_or_include_articles', 0);
         $filterInclude          = true;
         $articlesList           = [];
+        $currentArticleId       = $input->get('id', 0, 'UINT');
+
+        $isArticleAndShouldExcluded = $params->get('exclude_current', 1) === 1
+            && $input->get('option') === 'com_content'
+            && $input->get('view') === 'article';
+
```
Comment (Quy) [inline modules/mod_articles/src/Helper/ArticlesHelper.php:224]:
```suggestion
            && $input->get('view') === 'article';
```

File: modules/mod_articles/src/Helper/ArticlesHelper.php [hunk 5]
```diff
@@ -216,27 +216,48 @@ public function getArticles(Registry $params, SiteApplication $app)
...
 
         $articlesListToProcess = $params->get('included_articles', '');
```
Comment (Quy) [inline modules/mod_articles/src/Helper/ArticlesHelper.php]:
```suggestion
            && $input->get('view') === 'article';

        $articlesListToProcess = $params->get('included_articles', '');
```

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on f4e8760b2c0b53473ad6768419d07d75d0891845<br /><br />Thank you!!!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/44213">issues.joomla.org/tracker/joomla-cms/44213</a>.</sub>

**Comment by LadySolveig** [comment]
I have tested this item :white_check_mark: successfully on f4e8760b2c0b53473ad6768419d07d75d0891845<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/44213">issues.joomla.org/tracker/joomla-cms/44213</a>.</sub>

**Comment by chmst** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/44213">issues.joomla.org/tracker/joomla-cms/44213</a>.</sub>

**Comment by Hackwar** [comment]
Thank you for your contribution @rdeutz!

**Pull request merged**